templates/network.epp

```diff
@@ -0,0 +1,22 @@
+<%- | $addresses, $interface | -%>
+# THIS FILE IS MANAGED BY PUPPET
+# based on https://dn42.dev/howto/wireguard
+[Match]
+Name=<%= $interface %>
+
+[Network]
+DHCP=no
+IPv6AcceptRA=false
+IPForward=yes
+
+# for networkd >= 244 KeepConfiguration stops networkd from
+# removing routes on this interface when restarting
+KeepConfiguration=yes
+
+<% $addresses.each |$address| { -%>
+[Address]
+<% $address.each |$key, $value| { -%>
+<%= $key %>=<%= $value %>
+<% } -%>
+
```

Comment:
I'd keep the whitespace at the top. This prevents an empty line at the bottom of the file
```suggestion
<% $addresses.each |$address| { -%>

[Address]
<% $address.each |$key, $value| { -%>
<%= $key %>=<%= $value %>
<% } -%>
```